Awful.apk/src/main/java/com/ferg/awfulapp/AwfulActivity.kt

```diff
@@ -40,6 +40,7 @@ abstract class AwfulActivity : AppCompatActivity(), AwfulPreferences.AwfulPrefer
     private var customActivityTitle: TextView? = null
     // TODO: this is a var but honestly why does any activity need to replace it with their own copy? It's a singleton - if they're doing it for the callbacks, just use the register method
     var mPrefs: AwfulPreferences = AwfulPreferences.getInstance()
+    var activityInitialized = false
```

Comment:
Can you move this into ``ForumsIndexActivity``? It might be good to have something like this here at some point, but this flag's being used to guard the lateinit stuff, and it's being set before those things are actually initialised. It _works_ fine as things are, I'm just worried about it introducing subtle bugs down the line

I'd put it after the navigation drawer is initialised in ``onCreate``, but before the ``updateNavDrawer`` call that relies on it. And can you stick a comment above that just says the components are all initialised at this point? Maybe things need reordering in there but this'll do for now